vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/overview/ProjectOverviewHandler.java

```diff
@@ -51,11 +52,20 @@ public void handle(ProjectOverviewLayout layout) {
       configureSearchDropbox();
       configureSelectionModeDialog();
       setProjectsToGrid();
+      setupSearchBar();
     }
   }
 
+  private void setupSearchBar() {
+    registeredProjectOverview.projectSearchField.setValueChangeMode(ValueChangeMode.LAZY);
+    registeredProjectOverview.projectSearchField
+        .addValueChangeListener(listener -> registeredProjectOverview.projectGrid.setItems(
+            query -> projectRepository.query(registeredProjectOverview.projectSearchField.getValue(),
+                query.getOffset(), query.getLimit()).stream()));
+  }
+
   private void setProjectsToGrid(){
-    registeredProjectOverview.projectGrid.setItems(query -> projectRepository.getAllPreviews(query.getOffset(), query.getLimit()).stream());
+    registeredProjectOverview.projectGrid.setItems(query -> projectRepository.query(query.getOffset(), query.getLimit()).stream());
```

Comment:
you can include the value of the searchbar directly here. This prevents you from having two separate locations in the code where the items of the grid are set/overwritten.